core/async/src/test_loop/delay_sender.rs

```diff
@@ -0,0 +1,69 @@
+use std::{sync::Arc, time::Duration};
+
+use crate::messaging;
+
+use super::multi_instance::IndexedDelaySender;
+
+/// Interface to send an event with a delay (in virtual time). It can be
```

Comment:
what is "virtual time"?

Reply:
Virtual time is the TestLoop's fake time. Suppose, for example, that the Client fetches new blocks every 400ms, and this is normally, in production, triggered by a timer kept by the ClientActor every 100ms to check the expiry condition. In TestLoop, instead of the ClientActor, we would register a periodic_timer that fires every 100ms virtual time, and construct the Client with a Clock provided by the TestLoop (not implemented yet as of this PR) which would return 100ms, 200ms, 300ms, 400ms, etc. depending on the virtual time. When the TestLoop runs, it will run handle all events at 0ms, and then increment the virtual time to the whatever next event at >0ms, say 100ms, then run that event at virtual time 100ms, etc. This is how it is able to support time-based logic, while being deterministic.